### ISSUE TITLE
Colon rule triggering with numeric keys

### ISSUE THREAD

**Issue by msalmonse** [realm/SwiftLint#2838]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
* Paste your configuration file:

```yml
# insert yaml contents here
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcode-select -p`)?
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
let foo = try! bar()
```

**Comment by msalmonse** [comment]
The colon rule triggers on `223: [ 15, ... ]`

warning: Colon Violation: Colons should be next to the identifier when specifying a type and next to the key in dictionary literals. (colon)

SwiftLint: 0.34.0
Xcode 11.0 Beta 5

This causes a violation:
``` swift
let testDict: [Int: [Int]] = [
            6:   [ 6, 22, 38, 54, 70, 86, 102, 118, 134, 150, 166, 182, 198, 214, 230, 246 ],
            223: [ 15, 31, 47, 63, 79, 95, 111, 127, 143, 159, 175, 191, 207, 223, 239, 255 ]
        ]
```

**Comment by msalmonse** [comment]
Looked like shit
